src/nunavut/lang/c/support/nunavut.j2

```diff
@@ -0,0 +1,502 @@
+/*
+ *                                                                                                            +-+ +-+
+ *  UAVCAN common serialization support routines.                                                             | | | |
+ *                                                                                                            \  -  /
+ *  This software is distributed under the terms of the MIT License.                                            ---
+ *                                                                                                               o
+ * +------------------------------------------------------------------------------------------------------------------+
+ */
+#ifndef NUNAVUT_SUPPORT_NUNAVUT_H_INCLUDED
+#define NUNAVUT_SUPPORT_NUNAVUT_H_INCLUDED
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include <string.h>
+#include <float.h>
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+typedef uint32_t NunavutUnsignedBitLength;
+typedef int32_t  NunavutSignedBitLength;
+typedef float    NunavutFloat32;
+typedef double   NunavutFloat64;
+
+// appropriate name?
+// do we typedef this, leave it anon, or enum NAME
+#define NUNAVUT_ERR_INVALID_BUF 1
+#define NUNAVUT_ERR_INVALID_LEN 2
+#define NUNAVUT_ERR_INVALID_OFFSET 3
+#define NUNAVUT_ERR_INVALID_TAG 4
+
+/// By default, this macro resolves to the standard assert(). The user can redefine this if necessary.
+/// To disable assertion checks completely, make it expand into `(void)(0)`.
+#ifndef NUNAVUT_ASSERT
```

Comment:
We want to eliminate dead code and use the new language options mechanism I added. I'll update my sandbox branch with an example of using `option_enable_serialization_asserts` instead of using the C-preprocessor.

Reply:
I just pushed an example of adding a "generate asserts" option:

https://github.com/thirtytwobits/nunavut/commit/69823ae4258af8cdf56b3307dca5dd1a1a7ef16f

Reply:
I may have assumed you were done for the night and been working on this myself. :clown_face:

Took your changes because they added doc tests and added the float options as well.